flink-core/src/main/java/org/apache/flink/configuration/DeploymentOptions.java

```diff
@@ -34,7 +34,13 @@
 			key("execution.target")
 					.stringType()
 					.noDefaultValue()
-					.withDescription("The deployment target for the execution, e.g. \"local\" for local execution.");
+					.withDescription(" The deployment target for the execution. " +
+							"This can take one of the following values:" +
+							"\"remote\", \"local\", \"yarn-per-job\", " +
+							"\"yarn-session\", \"yarn-application\", " +
+							"\"kubernetes-session\" and \"kubernetes-application\". " +
+							"Note that \"yarn-application\" and \"kubernetes-application\" " +
+							"can only be used through the CLI using the \"run-application -t\" command.");
```

Comment:
Would it make sense to explain a bit the requirements for each deployment target? E.g. when using `yarn-session` I guess one also need to specify the `yarn.application.id`. The same for `kubernetes-session`. Or is it somewhere documented?

Reply:
I think that it is not that easy to do it here and also 1) for yarn there are a lot of combinations, for example do we have a `yarn.properties` (I do not remember the exact name) file or not, and 2) I am not sure about `Kubernetes`, what is required and what not.

Reply:
Do we have a concrete plan how to update this documentation? I think this would be a follow-up of the deployment target introduction. I think this quite important since this is what users will face quite soon after starting working with Flink.